### ISSUE TITLE
False positive when using "reduce" with list of strings

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#8632]
**Describe the bug**
After update from `1.1.373` to `1.1.374` i noticed this false positive:

```python
from typing import Any
from functools import reduce
from operator import getitem


def deep_getitem(data: dict[str, Any], attr: str) -> Any:
    """
    >>> deep_getitem({"data": {"key": 1}}, "data.key")
    1
    """
    return reduce(getitem, attr.split("."), data)
```

```
/home/serg/src/pyright-examples/reduce.py
  /home/serg/src/pyright-examples/reduce.py:11:28 - error: Argument of type "list[str]" cannot be assigned to parameter "sequence" of type "Iterable[_S@reduce]" in function "reduce"
    "list[str]" is incompatible with "Iterable[slice]"
      Type parameter "_T_co@Iterable" is covariant, but "str" is not a subtype of "slice"
        "str" is incompatible with "slice" (reportArgumentType)
  /home/serg/src/pyright-examples/reduce.py:11:45 - error: Argument of type "dict[str, Any]" cannot be assigned to parameter "initial" of type "_T@reduce" in function "reduce"
    "dict[str, Any]" is incompatible with "Sequence[_T@getitem]" (reportArgumentType)
2 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**

- pyright 1.1.374

**Comment by erictraut** [comment]
This is included in pyright 1.1.375.
